<!DOCTYPE html>
<html lang="en">
<head prefix="og: http://ogp.me/ns# article: http://ogp.me/ns/article# website: http://ogp.me/ns/website#">
    <meta http-equiv="content-type" content="text/html; charset=utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
    <meta name="description" content="">
    <meta property="og:title" content="alignment">
    
    <meta property="og:type" content="article">
    <meta property="article:published_time" content="2018-12-12">
    
    <meta property="og:description" content="">
    <meta property="og:url" content="/post/alignment/">
    <meta property="og:site_name" content="My Blog Site">
    
    <meta property="og:tags" content="align">
    
    <meta property="og:tags" content="DNA">
    
    <meta property="og:tags" content="RNA-seq">
    
    <meta property="og:tags" content="HISAT2">
    
    <meta name="generator" content="Hugo 0.55.6" />
    <title>alignment &middot; My Blog Site</title>
    
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.7/styles/default.min.css">
    
    <link rel="stylesheet" href="../../css/style.css">
    
    <link href="../../index.xml" rel="alternate" type="application/rss+xml" title="My Blog Site" />
    
    
    <link rel="icon" href="../../favicon.ico" />
    
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
    tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
    }
    });
    </script>
    <script async type="text/javascript"
      src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    </script>
    
    
</head>
<body>

<nav class="navbar navbar-default navbar-fixed-top visible-xs">
	<div class="container-fluid">
		<div class="navbar-header">
			<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
				<span class="sr-only">Toggle navigation</span>
				<span class="icon-bar"></span>
				<span class="icon-bar"></span>
				<span class="icon-bar"></span>
			</button>
			
				<a class="navbar-brand" href="../../">Qihao Wei&#39;s Home</a>
			
		</div>
		<div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
			<ul class="nav navbar-nav">
				
				
					<li><a href="../../about">About</a></li>
				
					<li><a href="../../tutorial/">Tutorials</a></li>
				
			</ul>
		</div>
	</div>
</nav>
<div class="container-fluid">
	<div class="row">
		<div id="menu" class="hidden-xs col-sm-4 col-md-3">
	<div id="menu-content" class="vertical-align">
		
			<h1 class="text-center"><a href="../../">Qihao Wei&#39;s Home</a></h1>
		
		
		
			<small class="text-center center-block">MY HUGO BLOG</small>
		
		
		
			<img id="profile-pic" src="//img/profile.png" alt="My Picture" class="img-circle center-block">
		
		<div id="social" class="text-center">
			
				<a href="https://github.com/qihaowei89"><i class="fa fa-github fa-2x"></i></a>
			
				<a href="https://twitter.com/alexanderurq"><i class="fa fa-twitter fa-2x"></i></a>
			
				<a href="https://linkedin.com"><i class="fa fa-linkedin fa-2x"></i></a>
			
				<a href="https://plus.google.com/&#43;YourID"><i class="fa fa-google-plus fa-2x"></i></a>
			

			<a href="mailto:weiqihao08@gmail.com"><i class="fa fa-envelope-o fa-2x"></i></a>
		</div>
		<div id="links" class="text-center">
			
			
				<a href="../../about">About</a>
			
				<a href="../../tutorial/">Tutorials</a>
			
		</div>
	</div>
</div>

		<div id="content" class="col-xs-12 col-sm-8 col-md-9">
			<div class="row">
				<div id="post" class="col-sm-offset-1 col-sm-10 col-md-10 col-lg-8">

<main>
	<header>
		<h1>alignment</h1>
	</header>

	<article>
		


<div id="hisat2" class="section level2">
<h2>Hisat2</h2>
<p>HISAT2是一个快速和敏感的比对程序，用于将下一代测序读数（全基因组、转录组和外显组测序数据）映射到人类基因组群体（以及单个参考基因组）。基于<code>BWT for graphs [Sirén et al. 2014]</code>的方法，设计和实现的一种独创的方法<code>graph FM index (GFM)</code>。除了使用代表群体的的全局GFM的index外，HISAT2还使用局部GFM的index，他们共同覆盖整个基因组，每个index代表56Kbp的区域，一共有55000个index。这种新的索引策略称为<code>层次graph FM index（HGFM）</code></p>
<p>HISAT2中包含人类参考基因组和1230万个常见SNPs位点的index大小为6.2G。SNPs包括1100万个单核苷酸多态性、72.8万缺失和55.5万个插入。这个索引中的插入和删除片段都很小(通常小于20bp)。计划将结构变化(SV)合并到这个索引中。</p>
<p><strong>Alignment summary</strong></p>
<p>summary for unpaired reads</p>
<pre><code>20000 reads; of these:
  20000 (100.00%) were unpaired; of these:
    1247 (6.24%) aligned 0 times
    18739 (93.69%) aligned exactly 1 time
    14 (0.07%) aligned &gt;1 times
93.77% overall alignment rate</code></pre>
<p>summary for paired reads</p>
<pre><code>10000 reads; of these:
  10000 (100.00%) were paired; of these:
    650 (6.50%) aligned concordantly 0 times
    8823 (88.23%) aligned concordantly exactly 1 time
    527 (5.27%) aligned concordantly &gt;1 times
    ----
    650 pairs aligned concordantly 0 times; of these:
      34 (5.23%) aligned discordantly 1 time
    ----
    616 pairs aligned 0 times concordantly or discordantly; of these:
      1232 mates make up the pairs; of these:
        660 (53.57%) aligned 0 times
        571 (46.35%) aligned exactly 1 time
        1 (0.08%) aligned &gt;1 times
96.70% overall alignment rate</code></pre>
<p><strong>性能优化</strong></p>
<p><code>-p</code>选项使HISAT2启动指定数量的并行线程,每个线程在不同的处理器/核心上运行,使比对吞吐量大约增加为线程数量倍数。</p>
<p><strong>命令行</strong></p>
<p><strong>设置函数选项</strong></p>
<p>一些HISAT2选项可以指定函数，而不是单个数字或设置。在这些情况下，用户须指定三个参数：(a)函数类型<strong>F</strong>，(b)常数项<strong>B</strong>和(c)系数**A 。可用的函数类型：常数(C)、线性(L)、平方根(S)和自然对数(G)。参数指定为<code>F,B,A</code>-即，函数类型、常数项，系数用逗号分隔，没有空格。常数项和系数可以是负数和/或浮点数。</p>
<p>例如：<br />
函数选项指定为L，-0.4，-0.6，表示函数：<code>f(x) = -0.4 + -0.6 * x</code><br />
函数选项指定为G, 1, 5.4, 表示函数：<code>f(x) = 1.0 + 5.4 * ln(x)</code></p>
<p><strong>用法</strong><br />
<code>hisat2 [options]* -x &lt;hisat2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --sra-acc &lt;SRA accession number&gt;} [-S &lt;sam&gt;]</code></p>
</div>

	</article>
</main>

<div id="bottom-nav" class="text-center center-block">
	<a href=" /" class="btn btn-default"><i class="fa fa-home"></i> Home</a>
</div>


  <div id="disqus_thread"></div>
<script type="application/javascript">
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "shortcode-here" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


						</div>
					</div>
				</div>
			</div>
		</div>
  </div>
  
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

    ga('create', 'code-here', 'auto');
    ga('send', 'pageview');
    window.baseURL = "\/";
  </script>
  
  <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.2/js/bootstrap.min.js"></script>
  
  <script src="//cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.min.js"></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/topojson/1.6.9/topojson.min.js"></script>
  
  
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.7/highlight.min.js"></script>
  
  <script src="//js/App.js"></script>
  
</body>
</html>
