---
title: 学习生物信息常用流程-1
author: ''
date: '2018-12-07'
slug: 学习生物信息常用流程-1
categories:
  - Bioinfomatics
tags:
  - RNA-seq 
  - pipeline     
  - bioinfomatics
css: []
description: ''
highlight: yes
index: no
scripts: []
---

### RNACocktail:一种精确高效的RNA-seq分析框架   

From：https://bioinform.github.io/rnacocktail/

RNACocktail pipeline可对短read和长read进行广谱的RNA-Seq分析，以便能够从转录数据获得有意义的见解。它是在分析了各种RNA-Seq样品(从生殖系、癌症到干细胞数据集)以及使用多种工具组合来确定一个全面、快速和准确的pipeline的技术之后开发的。   


|short-read|long-read|
|----------|---------|
|alignment|error correction|
|transcriptome reconstruction|alignment|
|denovo transcriptome assembly|transcriptome reconstruction|
|alignment-free quantification|fusion prediction|
|differential expression analysis||	
|fusion prediction||	
|variant calling||	
|RNA editing prediction||


### 1 下载RNACocktail   
[最新版本下载地址](https://github.com/bioinform/RNACocktail/archive/v0.2.2.tar.gz)   
[历史版本下载地址](https://github.com/bioinform/RNACocktail/releases)

### 2 RNACocktail Docker镜像   

Docker镜像的[下载地址](https://hub.docker.com/r/marghoob/rnacocktail/)   


### 3 系统要求   
安装Python 2.7 以及下面的py包：

|Tool	|Version tested|Pipeline modes used in|
|:-----:|:-------:|:------:|
|pybedtools|0.7.7|editing|
|pysam|	0.9.0|	editing|
|numpy|	1.12.0|	editing|
|scipy|	0.18.1|	long_fusion|
|biopython|	1.66|	fusion|
|openpyxl|	1.5.6|	fusion|
|pandas|	0.19.2|	fusion|
|xlrd|0.6.1|	fusion|

pybedtools依赖[bedtools](https://github.com/arq5x/bedtools2)(v2.27.0) 软件。   

此外，必须提供以下工具的路径作为RNACocktail参数。或者，工具的可执行文件可以是环境变量中路径，也可以定义在`defaults.py`：   


   

|Tool|Version|Pipeline modes used in|
|----|-----------------|-----------------------------|
|[SAMtools](https://github.com/samtools/samtools)|1.2|align, reconstruct, long_align, long_reconstruct, and editing|
|[HISAT2](http://ccb.jhu.edu/software/hisat2/index.shtml)|2.0.5|align|
|[StringTie](https://ccb.jhu.edu/software/stringtie/index.shtml)|1.3.3|reconstruct and diff|
|[Salmon](https://github.com/COMBINE-lab/salmon)|0.8.0|quantify
|[Oases](https://github.com/dzerbino/oases)|0.2.09|assembly
|[Velvet](https://github.com/dzerbino/velvet)|1.2.10|assembly
|R with [DESeq2](https://www.bioconductor.org/packages/devel/bioc/html/DESeq2.html), [readr](https://github.com/hadley/readr), and [tximport](https://github.com/mikelove/tximport),readr,rjson, libraries|3.3.2|diff, editing
|[featureCounts](http://bioinf.wehi.edu.au/featureCounts/)|1.5.0-p1|diff
|[LoRDEC](http://www.atgc-montpellier.fr/lordec/)|0.6|long_correct
|[STAR](https://github.com/alexdobin/STAR)|2.5.2b|long_align
|[IDP](https://github.com/bioinform/IDP/)|0.1.9|long_reconstruct
|[IDP-fusion](https://www.healthcare.uiowa.edu/labs/au/IDP-fusion/default.asp)|1.1.1|long_fusion
|[GATK](https://software.broadinstitute.org/gatk/)|3.5-0|variant and editing
|[Picard](https://broadinstitute.github.io/picard/)|2.2.2|variant
|[GIREMI](https://github.com/zhqingit/giremi)|0.2.1|editing
|[gatb-core](https://github.com/GATB/gatb-core)|1.1.0|editing
|[HTSlib](https://github.com/samtools/htslib)|1.3|editing
|[FusionCatcher](https://github.com/ndaniel/fusioncatcher)|0.99.5a beta|fusion
|[bowtie](http://bowtie-bio.sourceforge.net/index.shtml)|1.2.0|fusion
|[bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)|2.2.9|fusion, long_fusion
|[bwa](http://bio-bwa.sourceforge.net/)|0.7.15|fusion
|[sra toolkit](https://github.com/ncbi/sra-tools)|2.8.1|fusion
|[coreutils](http://ftp.gnu.org/gnu/coreutils/)|8.25|fusion
|[pigz](http://zlib.net/pigz/)|2.3.1|fusion
|[blat](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/blat)|0.35|fusion
|[faToTwoBit](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/faToTwoBit)||fusion
|[liftOver](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64.v287/liftOver)||fusion
|[SeqTK](https://github.com/ndaniel/seqtk)|1.0-r82b|fusion
|[gmap](http://research-pub.gene.com/gmap/)|2017-02-15|long_fusion|   

### 4 安装RNACocktail   
RNACocktail是python的一个软件包，可以使用pip安装(https://github.com/bioinform/RNACocktail/archive/v0.2.2.tar.gz),通常，安装源将是:https://github.com/bioinform/RNACocktail/archive/version.tar.gz。   

### 5 运行RNACocktail   

`run_rnacocktail.py -h`   get  help      
`run_rnacocktail.py align -h`   get   short-read alignment help.   
`run_rnacocktail.py reconstruct -h`  get short-read transcriptome reconstruction help.   
`run_rnacocktail.py quantify -h`  get  short-read quantification help.   
`run_rnacocktail.py diff -h`  get  short-read differential expression help.   
`run_rnacocktail.py denovo -h`  get short-read de novo assembly help.   
`run_rnacocktail.py long_correct -h`  get long-read error correction help.   
`run_rnacocktail.py long_align -h`  get long-read alignment help.   
`run_rnacocktail.py long_reconstruct -h`  get long-read transcriptome reconstruction help.   
`run_rnacocktail.py long_fusion -h`  get long-read fusion detection help.   
`run_rnacocktail.py variant -h`  get variant calling help.   
`run_rnacocktail.py editing -h` get RNA editing detection help.   
`run_rnacocktail.py fusion -h` get RNA fusion detection help.   
`run_rnacocktail.py all -h` get  running all RNACocktail pipeline steps help.    

RNACocktail的all模式会根据输入数据，自动执行尽可能全面的分析，包括从比对到差异表达式分析的等步骤。   


### 6 RNACoktail测试   
**Small test**   
`cd test`   
`./test_run.sh`   

**Extensive test of all modes on Docker image**   
`cd test`   
`./docker_test.sh`   


### 7 分析脚本   
分析用到的脚本可以在`analaysis_scripts`文件夹中找到   

### 8 输出文件    

下表总结了各个RNACocktail模式下可能生成的输出文件。   

|Task|Command|Default Tool|Output Files|
|------------------------------|----------------------|---------------------------|--------------------------------------------|
Short-read alignment|align|HISAT2|**alignments:**alignments.sorted.bam **junctions:**splicesites.tab
Short-read transcriptome reconstruction|reconstruct|StringTie|**trasncripts:**transcripts.gtf  **expressions:**gene_abund.tab
Short-read quantification|quantify|Salmon-SMEM|**expressions:**quant.sf
Short-read differential expression|diff|DESeq2|**differential expressions:**deseq2_res.tab
Short-read de novo assembly|denovo|Oases|**trasncripts:**transcripts.fa
Long-read error correction|long_correct|LoRDEC|**corrected reads:**long_corrected.fa
Long-read alignment|long_align|STARlong|**alignments:**Aligned.out.psl
Long-read transcriptome reconstruction|long_reconstruct|IDP|**trasncripts:**isoform.gtf **expressions:**isoform.exp
Long-read fusion detection|long_fusion|IDP-fusion|**fusions:**fusion_report.tsv
Variant calling|variant|GATK|**variants:**variants_filtered.vcf
RNA editing detection|editing|GIREMI|**edits:**giremi_out.txt.res
RNA Fusion detection|fusion|FusionCatcher|**fusions:**final-list_candidate-fusion-genes.txt
Running all steps|all|whole pipeline|all outputs of the successful steps.


### 9 Example    
接下来介绍用于以各种模式和数据类型（short-reads和long-reads）运行RNACocktail的一些Example 命令行。特别地，Example 17和18显示了如何使用`all`模式进行最全面的分析。注意RNACocktail需要预先构建参考基因组和参考转录组的索引。   

**Example  1 (align short paired-end):**   
Run of RNACocktail for alignment of paired-end short-read sequences (HISAT2).   

```
run_rnacocktail.py align \
      --align_idx hisat2-idx \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --1 seq_1.fq.gz \
      --2 seq_2.fq.gz \
      --hisat2 /path/to/hisat2 \
      --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
      --samtools /path/to/samtools \
      --threads 10 \
      --sample A
```   
**Example  2 (align short single-end):**    
Run of RNACocktail for alignment of single-end short-read sequences (HISAT2).
```
run_rnacocktail.py align \
      --align_idx hisat2-idx \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --U seq.fq.gz \
      --hisat2 /path/to/hisat2 \
      --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
      --samtools /path/to/samtools \
      --threads 10 \
      --sample A
```   

**Example  3 (reconstruct short-read):**    
Run of RNACocktail for short-read transcriptome reconstruction (StringTie).  
```
run_rnacocktail.py reconstruct \
      --alignment_bam work/hisat2/A/alignments.sorted.bam \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --stringtie /path/to/stringtie \
      --threads 10 \
      --sample A
```   

**Example  4 (quantify short paired-end):(alignment-free)**    
Run of RNACocktail for (alignment-free) quantification of paired-end short-read sequences (Salmon-SMEM). 
```
salmon index \
    -t Homo_sapiens.GRCh38.cdna.1.fa \
    -i Homo_sapiens.GRCh38.cdna.1.quasi \
    --type quasi
    
for sample in A1 A2 B1 B2
do 
run_rnacocktail.py quantify 
      --quantifier_idx Homo_sapiens.GRCh38.cdna.1.quasi \
      --1 ../${sample}_1.fq.gz \
      --2 ../${sample}_2.fq.gz \
      --libtype IU \
      --salmon_k 19 \
      --outdir out \
      --workdir work \
      --salmon salmon \
      --threads 4 \
      --sample A \
      --unzip
done   


```   

**Example  5 (quantify short single-end):(alignment-free)**    
Run of RNACocktail for (alignment-free) quantification of single-end short-read sequences (Salmon-SMEM).
```
salmon index \
    -t Homo_sapiens.GRCh38.cdna.1.fa \
    -i Homo_sapiens.GRCh38.cdna.1.quasi \
    --type quasi


for sample in A1 A2 B1 B2
do  
run_rnacocktail.py quantify \
  --quantifier_idx Homo_sapiens.GRCh38.cdna.1.quasi \
  --U ../${sample}.fq.gz \
  --libtype IU \
  --salmon_k 19 \
  --outdir out  \
  --workdir work \
  --threads 4 \
  --sample ${sample} \
  --unzip 
done
 
```   


**Example  6 (diff alignment-free):**   
Run of RNACocktail for differential expression analysis of quantifications computed using Salmon-SMEM (DESeq2).
```
run_rnacocktail.py diff \
    --quant_files \
      work/salmon_smem/A1/quant.sf,work/salmon_smem/A2/quant.sf \
      work/salmon_smem/B1/quant.sf,work/salmon_smem/B2/quant.sf \
    --sample A1,A2 B1,B2 \
    --ref_gtf Homo_sapiens.GRCh38.90.chr.1.gtf \
    --outdir out \
    --workdir work
```   

**Example  7 (diff alignment):**   
Run of RNACocktail for differential expression analysis of reads aligned using HISAT2 on reference transcriptome (DESeq2).   

```
run_rnacocktail.py diff \
    --alignments \
      work/hisat2/A1/alignments.sorted.bam,work/hisat2/A2/alignments.sorted.bam \
      work/hisat2/B1/alignments.sorted.bam,work/hisat2/B2/alignments.sorted.bam \
    --sample A1,A2 B1,B2 \
    --ref_gtf genes.GRCh37.gtf \
    --outdir out \
    --workdir work \
    --featureCounts /path/to/featureCounts
```   

**Example 8 (diff):**   
Run of RNACocktail for differential expression analysis of reads aligned using HISAT2 on StringTie computed transcriptome (DESeq2).   
```
run_rnacocktail.py diff \
    --alignments \
      work/hisat2/A1/alignments.sorted.bam,work/hisat2/A2/alignments.sorted.bam \
      work/hisat2/B1/alignments.sorted.bam,work/hisat2/B2/alignments.sorted.bam \
    --transcripts_gtfs work/stringtie/A1/transcripts.gtf,work/stringtie/A2/transcripts.gtf \
      work/stringtie/B1/transcripts.gtf,work/stringtie/B2/transcripts.gtf \
    --sample A1,A2 B1,B2 \
    --ref_gtf genes.GRCh37.gtf \
    --outdir out \
    --workdir work \
    --featureCounts /path/to/featureCounts
```   
**Example 9 (denovo):**   
Run of RNACocktail for de novo assembly (Oases).   
```
run_rnacocktail.py denovo \
    --1 seq_1.fq.gz \
    --2 seq_2.fq.gz \
    --outdir out \
    --workdir work \
    --oases /path/to/oases \
    --velveth /path/to/velveth \
    --velvetg /path/to/velvetg \
    --threads 4 \
    --sample A \
    --file_format fastq.gz
```   

**Example 10 (long_correct):**   
Run of RNACocktail for long-read error correction (LoRDEC).   
```
run_rnacocktail.py long_correct \
    --kmer 23 \
    --solid 3 \
    --short seq.fq.gz \
    --long seq_long.fa \
    --outdir out \
    --workdir work \
    --lordec /path/to/lordec-correct \
    --threads 4 \
    --sample A
```   

**Example 11 (long_align):**   
Run of RNACocktail for long-read alignment (STARlong).   
```
run_rnacocktail.py long_align \
    --long work/lordec/A/long_corrected.fa \
    --outdir out \
    --workdir work \
    --starlong /path/to/STARlong \
    --threads 4 \
    --sample A \
    --sam2psl /path/to/sam2psl.py \
    --samtools /path/to/samtools -\
    -genome_dir /path/to/STAR/genome_idx
```   
**Example 12 (long_reconstruct):**   

Run of RNACocktail for long-read transcriptome reconstruction (IDP).   
```
run_rnacocktail.py long_reconstruct \
    --alignment work/hisat2/A/alignments.sorted.bam \
    --short_junction work/hisat2/A/splicesites.bed \
    --long_alignment work/starlong/A/Aligned.out.psl \
    --outdir out \
    --workdir work \
    --idp /path/to/runIDP.py \
    --threads 4 \
    --sample A \
    --read_length 100 \
    --ref_genome genome.GRCh37.fa \
    --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd \
    --ref_gpd genes.GRCh37.refFlat.txt \
    --samtools /path/to/samtools \
    --idp_cfg idp.cfg
```    

**Example 13 (long_fusion):**   
Run of RNACocktail for long-read fusion detection (IDP-fusion).   
```
run_rnacocktail.py long_fusion \
    --alignment work/hisat2/A/alignments.sorted.bam \
    --short_junction work/hisat2/A/splicesites.bed \
    --short_fasta seq.fa \
    --long_fasta work/lordec/A/long_corrected.fa \
    --outdir out \
    --workdir work \
    --threads 4 \
    --sample A \
    --ref_genome genome.GRCh37.fa \
    --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd \
    --ref_gpd genes.GRCh37.refFlat.txt \
    --read_length 100 \
    --genome_bowtie2_idx genome.bt2_idx \
    --transcriptome_bowtie2_idx genes.bt2_idx \
    --uniqueness_bedgraph uniqueness.bedGraph \
    --gmap_idx gmap_idx \
    --idpfusion /path/to/runIDP.py \
    --samtools /path/to/samtools \
    --idpfusion_cfg idpfusion.cfg
```   

Example 14 (variant):
Run of RNACocktail for RNA-Seq variant calling (GATK) with enabled IndelRealignment.
run_rnacocktail.py variant --alignment work/hisat2/A/alignments.sorted.bam --outdir out --workdir work --picard /path/to/picard.jar --gatk /path/to/GenomeAnalysisTK.jar --threads 10 --sample A --ref_genome genome.GRCh37.fa --IndelRealignment --knownsites dbsnp_138.b37.vcf

Example 15 (editing):
Run of RNACocktail for RNA editing detection (GIREMI)
run_rnacocktail.py editing --alignment work/gatk/A/bsqr.bam --variant work/gatk/A/variants_filtered.vcf --strand_pos test/GRCh37_strand_pos.bed --genes_pos test/GRCh37_genes_pos.bed --outdir out --workdir work --giremi_dir /path/to/giremi/directory/ --gatk /path/to/GenomeAnalysisTK.jar --samtools /path/to/samtools --htslib_dir /path/to/htslib/directory/ --threads 10 --sample A --ref_genome genome.GRCh37.fa --knownsites dbsnp_138.b37.vcf

Example 16 (fusion):
Run of RNACocktail for RNA fusion detection (FusionCatcher)
run_rnacocktail.py fusion --data_dir /path/to/fusioncatcher/ensembl/data/directory/ --input seq_1.fq.gz,seq_2.fq.gz --outdir out --workdir work --fusioncatcher /path/to/fusioncatcher --threads 4 --sample A

Example 17 (all):
Run all pipeline steps (Short-read example)
run_rnacocktail.py all --outdir out --workdir work --threads 10 --1 A1_1.fq.gz,A2_1.fq.gz B1_1.fq.gz,B2_1.fq.gz --2 A1_2.fq.gz,A2_2.fq.gz B1_2.fq.gz,B2_2.fq.gz --sample all_A1,all_A2 all_B1,all_B2 --ref_gtf genes.GRCh37.gtf --ref_genome genome.GRCh37.fa --align_idx hisat2-idx --quantifier_idx salmon_fmd_idx --unzip --file_format fastq.gz --IndelRealignment --CleanSam --knownsites dbsnp_138.b37.vcf --strand_pos test/GRCh37_strand_pos.bed --genes_pos test/GRCh37_genes_pos.bed --data_dir /path/to/fusioncatcher/ensembl/data/directory/ --giremi_dir /path/to/giremi/directory/ --gatk /path/to/GenomeAnalysisTK.jar --htslib_dir /path/to/htslib/directory/ --picard /path/to/picard.jar --samtools /path/to/samtools --hisat2 /path/to/hisat2 --hisat2_sps /path/to/hisat2_extract_splice_sites.py --stringtie /path/to/stringtie --salmon /path/to/salmon --featureCounts /path/to/featureCounts --oases /path/to/oases --velveth /path/to/velveth --velvetg /path/to/velvetg --lordec /path/to/lordec-correct --sam2psl /path/to/sam2psl.py --fusioncatcher /path/to/fusioncatcher

Example 18 (all):
Run all pipeline steps (long-read example)
run_rnacocktail.py all --outdir out --workdir work --threads 10 --U seq_short.fa --long seq_long.fa --sample all_C --ref_gtf genes.GRCh37.gtf --ref_genome genome.GRCh37.fa --align_idx hisat2-idx --quantifier_idx salmon_fmd_idx --unzip --file_format fasta --IndelRealignment --CleanSam --knownsites dbsnp_138.b37.vcf --strand_pos test/GRCh37_strand_pos.bed --genes_pos test/GRCh37_genes_pos.bed --data_dir /path/to/fusioncatcher/ensembl/data/directory/ --giremi_dir /path/to/giremi/directory/ --gatk /path/to/GenomeAnalysisTK.jar --htslib_dir /path/to/htslib/directory/ --star_genome_dir /path/to/STAR/genome_idx/ --genome_bowtie2_idx genome.bt2_idx --transcriptome_bowtie2_idx genes.bt2_idx --uniqueness_bedgraph uniqueness.bedGraph --gmap_idx gmap_idx --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd --ref_gpd genes.GRCh37.refFlat.txt --read_length 100 --picard /path/to/picard.jar --hisat2_opts \"-f\" --idp /path/to/idp/runIDP.py --idpfusion /path/to/idpfusion/runIDP.py --samtools /path/to/samtools --hisat2 /path/to/hisat2 --hisat2_sps /path/to/hisat2_extract_splice_sites.py --stringtie /path/to/stringtie --salmon /path/to/salmon --featureCounts /path/to/featureCounts --oases /path/to/oases --velveth /path/to/velveth --velvetg /path/to/velvetg --lordec /path/to/lordec-correct --sam2psl /path/to/sam2psl.py --fusioncatcher /path/to/fusioncatcher



















