---
title: 学习生物信息常用流程-1
author: ''
date: '2018-12-07'
slug: 学习生物信息常用流程-1
categories:
  - Bioinfomatics
tags:
  - RNA-seq 
  - pipeline     
  - bioinfomatics
css: []
description: ''
highlight: yes
index: no
scripts: []
---

### RNACocktail:一种精确高效的RNA-seq分析框架   

From：https://bioinform.github.io/rnacocktail/

RNACocktail pipeline可对短read和长read进行广谱的RNA-Seq分析，以便能够从转录数据获得有意义的见解。它是在分析了各种RNA-Seq样品(从生殖系、癌症到干细胞数据集)以及使用多种工具组合来确定一个全面、快速和准确的pipeline的技术之后开发的。   


|short-read|long-read|
|----------|---------|
|alignment|error correction|
|transcriptome reconstruction|alignment|
|denovo transcriptome assembly|transcriptome reconstruction|
|alignment-free quantification|fusion prediction|
|differential expression analysis||	
|fusion prediction||	
|variant calling||	
|RNA editing prediction||


### 1 下载RNACocktail   
[最新版本下载地址](https://github.com/bioinform/RNACocktail/archive/v0.2.2.tar.gz)   
[历史版本下载地址](https://github.com/bioinform/RNACocktail/releases)

### 2 RNACocktail Docker镜像   

Docker镜像的[下载地址](https://hub.docker.com/r/marghoob/rnacocktail/)   


### 3 系统要求   
安装Python 2.7 以及下面的py包：

|Tool	|Version tested|Pipeline modes used in|
|:-----:|:-------:|:------:|
|pybedtools|0.7.7|editing|
|pysam|	0.9.0|	editing|
|numpy|	1.12.0|	editing|
|scipy|	0.18.1|	long_fusion|
|biopython|	1.66|	fusion|
|openpyxl|	1.5.6|	fusion|
|pandas|	0.19.2|	fusion|
|xlrd|0.6.1|	fusion|

pybedtools依赖[bedtools](https://github.com/arq5x/bedtools2)(v2.27.0) 软件。   

此外，必须提供以下工具的路径作为RNACocktail参数。或者，工具的可执行文件可以是环境变量中路径，也可以定义在`defaults.py`：   


|Tool|Version|Pipeline modes used in|
|----|-----------------|-----------------------------|
|[SAMtools](https://github.com/samtools/samtools)|1.2|align, reconstruct, long_align, long_reconstruct, and editing|
|[HISAT2](http://ccb.jhu.edu/software/hisat2/index.shtml)|2.0.5|align|
|[StringTie](https://ccb.jhu.edu/software/stringtie/index.shtml)|1.3.3|reconstruct and diff|
|[Salmon](https://github.com/COMBINE-lab/salmon)|0.8.0|quantify
|[Oases](https://github.com/dzerbino/oases)|0.2.09|assembly
|[Velvet](https://github.com/dzerbino/velvet)|1.2.10|assembly
|R with [DESeq2](https://www.bioconductor.org/packages/devel/bioc/html/DESeq2.html), [readr](https://github.com/hadley/readr), and [tximport](https://github.com/mikelove/tximport),readr,rjson, libraries|3.3.2|diff, editing
|[featureCounts](http://bioinf.wehi.edu.au/featureCounts/)|1.5.0-p1|diff
|[LoRDEC](http://www.atgc-montpellier.fr/lordec/)|0.6|long_correct
|[STAR](https://github.com/alexdobin/STAR)|2.5.2b|long_align
|[IDP](https://github.com/bioinform/IDP/)|0.1.9|long_reconstruct
|[IDP-fusion](https://www.healthcare.uiowa.edu/labs/au/IDP-fusion/default.asp)|1.1.1|long_fusion
|[GATK](https://software.broadinstitute.org/gatk/)|3.5-0|variant and editing
|[Picard](https://broadinstitute.github.io/picard/)|2.2.2|variant
|[GIREMI](https://github.com/zhqingit/giremi)|0.2.1|editing
|[gatb-core](https://github.com/GATB/gatb-core)|1.1.0|editing
|[HTSlib](https://github.com/samtools/htslib)|1.3|editing
|[FusionCatcher](https://github.com/ndaniel/fusioncatcher)|0.99.5a beta|fusion
|[bowtie](http://bowtie-bio.sourceforge.net/index.shtml)|1.2.0|fusion
|[bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)|2.2.9|fusion, long_fusion
|[bwa](http://bio-bwa.sourceforge.net/)|0.7.15|fusion
|[sra toolkit](https://github.com/ncbi/sra-tools)|2.8.1|fusion
|[coreutils](http://ftp.gnu.org/gnu/coreutils/)|8.25|fusion
|[pigz](http://zlib.net/pigz/)|2.3.1|fusion
|[blat](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/blat)|0.35|fusion
|[faToTwoBit](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/faToTwoBit)||fusion
|[liftOver](http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64.v287/liftOver)||fusion
|[SeqTK](https://github.com/ndaniel/seqtk)|1.0-r82b|fusion
|[gmap](http://research-pub.gene.com/gmap/)|2017-02-15|long_fusion|   

### 4 安装RNACocktail   
RNACocktail是python的一个软件包，可以使用pip安装(https://github.com/bioinform/RNACocktail/archive/v0.2.2.tar.gz),通常，安装源将是:https://github.com/bioinform/RNACocktail/archive/version.tar.gz。   

### 5 运行RNACocktail   

`run_rnacocktail.py -h`   get  help      
`run_rnacocktail.py align -h`   get   short-read alignment help.   
`run_rnacocktail.py reconstruct -h`  get short-read transcriptome reconstruction help.   
`run_rnacocktail.py quantify -h`  get  short-read quantification help.   
`run_rnacocktail.py diff -h`  get  short-read differential expression help.   
`run_rnacocktail.py denovo -h`  get short-read de novo assembly help.   
`run_rnacocktail.py long_correct -h`  get long-read error correction help.   
`run_rnacocktail.py long_align -h`  get long-read alignment help.   
`run_rnacocktail.py long_reconstruct -h`  get long-read transcriptome reconstruction help.   
`run_rnacocktail.py long_fusion -h`  get long-read fusion detection help.   
`run_rnacocktail.py variant -h`  get variant calling help.   
`run_rnacocktail.py editing -h` get RNA editing detection help.   
`run_rnacocktail.py fusion -h` get RNA fusion detection help.   
`run_rnacocktail.py all -h` get  running all RNACocktail pipeline steps help.    

RNACocktail的all模式会根据输入数据，自动执行尽可能全面的分析，包括从比对到差异表达式分析的等步骤。   


### 6 RNACoktail测试   
**Small test**   
`cd test`   
`./test_run.sh`   

**Extensive test of all modes on Docker image**   
`cd test`   
`./docker_test.sh`   


### 7 分析脚本   
分析用到的脚本可以在`analaysis_scripts`文件夹中找到   

### 8 输出文件    

下表总结了各个RNACocktail模式下可能生成的输出文件。   

|Task|Command|Default Tool|Output Files|
|------------------------------|----------------------|---------------------------|--------------------------------------------|
Short-read alignment|align|HISAT2|**alignments:**alignments.sorted.bam **junctions:**splicesites.tab
Short-read transcriptome reconstruction|reconstruct|StringTie|**trasncripts:**transcripts.gtf  **expressions:**gene_abund.tab
Short-read quantification|quantify|Salmon-SMEM|**expressions:**quant.sf
Short-read differential expression|diff|DESeq2|**differential expressions:**deseq2_res.tab
Short-read de novo assembly|denovo|Oases|**trasncripts:**transcripts.fa
Long-read error correction|long_correct|LoRDEC|**corrected reads:**long_corrected.fa
Long-read alignment|long_align|STARlong|**alignments:**Aligned.out.psl
Long-read transcriptome reconstruction|long_reconstruct|IDP|**trasncripts:**isoform.gtf **expressions:**isoform.exp
Long-read fusion detection|long_fusion|IDP-fusion|**fusions:**fusion_report.tsv
Variant calling|variant|GATK|**variants:**variants_filtered.vcf
RNA editing detection|editing|GIREMI|**edits:**giremi_out.txt.res
RNA Fusion detection|fusion|FusionCatcher|**fusions:**final-list_candidate-fusion-genes.txt
Running all steps|all|whole pipeline|all outputs of the successful steps.


### 9 Example    
接下来介绍用于以各种模式和数据类型（short-reads和long-reads）运行RNACocktail的一些Example 命令行。特别地，Example 17和18显示了如何使用`all`模式进行最全面的分析。注意RNACocktail需要预先构建参考基因组和参考转录组的索引。   

**Example  1 (align short paired-end):**   
Run of RNACocktail for alignment of paired-end short-read sequences (HISAT2).   

```
run_rnacocktail.py align \
      --align_idx hisat2-idx \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --1 seq_1.fq.gz \
      --2 seq_2.fq.gz \
      --hisat2 /path/to/hisat2 \
      --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
      --samtools /path/to/samtools \
      --threads 10 \
      --sample A
```   
**Example  2 (align short single-end):**    
Run of RNACocktail for alignment of single-end short-read sequences (HISAT2).
```
run_rnacocktail.py align \
      --align_idx hisat2-idx \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --U seq.fq.gz \
      --hisat2 /path/to/hisat2 \
      --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
      --samtools /path/to/samtools \
      --threads 10 \
      --sample A
```   

**Example  3 (reconstruct short-read):**    
Run of RNACocktail for short-read transcriptome reconstruction (StringTie).  
```
run_rnacocktail.py reconstruct \
      --alignment_bam work/hisat2/A/alignments.sorted.bam \
      --outdir out \
      --workdir work \
      --ref_gtf genes.GRCh37.gtf \
      --stringtie /path/to/stringtie \
      --threads 10 \
      --sample A
```   

**Example  4 (quantify short paired-end):(alignment-free)**    
Run of RNACocktail for (alignment-free) quantification of paired-end short-read sequences (Salmon-SMEM). 
```
salmon index \
    -t Homo_sapiens.GRCh38.cdna.1.fa \
    -i Homo_sapiens.GRCh38.cdna.1.quasi \
    --type quasi
    
for sample in A1 A2 B1 B2
do 
run_rnacocktail.py quantify 
      --quantifier_idx Homo_sapiens.GRCh38.cdna.1.quasi \
      --1 ../${sample}_1.fq.gz \
      --2 ../${sample}_2.fq.gz \
      --libtype IU \
      --salmon_k 19 \
      --outdir out \
      --workdir work \
      --salmon salmon \
      --threads 4 \
      --sample A \
      --unzip
done   

```   

**Example  5 (quantify short single-end):(alignment-free)**    
Run of RNACocktail for (alignment-free) quantification of single-end short-read sequences (Salmon-SMEM).
```
salmon index \
    -t Homo_sapiens.GRCh38.cdna.1.fa \
    -i Homo_sapiens.GRCh38.cdna.1.quasi \
    --type quasi


for sample in A1 A2 B1 B2
do  
run_rnacocktail.py quantify \
  --quantifier_idx Homo_sapiens.GRCh38.cdna.1.quasi \
  --U ../${sample}.fq.gz \
  --libtype IU \
  --salmon_k 19 \
  --outdir out  \
  --workdir work \
  --threads 4 \
  --sample ${sample} \
  --unzip 
done
 
```   


**Example  6 (diff alignment-free):**   
Run of RNACocktail for differential expression analysis of quantifications computed using Salmon-SMEM (DESeq2).
```
run_rnacocktail.py diff \
    --quant_files \
      work/salmon_smem/A1/quant.sf,work/salmon_smem/A2/quant.sf \
      work/salmon_smem/B1/quant.sf,work/salmon_smem/B2/quant.sf \
    --sample A1,A2 B1,B2 \
    --ref_gtf Homo_sapiens.GRCh38.90.chr.1.gtf \
    --outdir out \
    --workdir work
```   

**Example  7 (diff alignment):**   
Run of RNACocktail for differential expression analysis of reads aligned using HISAT2 on reference transcriptome (DESeq2).   

```
run_rnacocktail.py diff \
    --alignments \
      work/hisat2/A1/alignments.sorted.bam,work/hisat2/A2/alignments.sorted.bam \
      work/hisat2/B1/alignments.sorted.bam,work/hisat2/B2/alignments.sorted.bam \
    --sample A1,A2 B1,B2 \
    --ref_gtf genes.GRCh37.gtf \
    --outdir out \
    --workdir work \
    --featureCounts /path/to/featureCounts
```   

**Example 8 (diff):**   
Run of RNACocktail for differential expression analysis of reads aligned using HISAT2 on StringTie computed transcriptome (DESeq2).   
```
run_rnacocktail.py diff \
    --alignments \
      work/hisat2/A1/alignments.sorted.bam,work/hisat2/A2/alignments.sorted.bam \
      work/hisat2/B1/alignments.sorted.bam,work/hisat2/B2/alignments.sorted.bam \
    --transcripts_gtfs work/stringtie/A1/transcripts.gtf,work/stringtie/A2/transcripts.gtf \
      work/stringtie/B1/transcripts.gtf,work/stringtie/B2/transcripts.gtf \
    --sample A1,A2 B1,B2 \
    --ref_gtf genes.GRCh37.gtf \
    --outdir out \
    --workdir work \
    --featureCounts /path/to/featureCounts
```   
**Example 9 (denovo):**   
Run of RNACocktail for de novo assembly (Oases).   
```
run_rnacocktail.py denovo \
    --1 seq_1.fq.gz \
    --2 seq_2.fq.gz \
    --outdir out \
    --workdir work \
    --oases /path/to/oases \
    --velveth /path/to/velveth \
    --velvetg /path/to/velvetg \
    --threads 4 \
    --sample A \
    --file_format fastq.gz
```   

**Example 10 (long_correct):**   
Run of RNACocktail for long-read error correction (LoRDEC).   
```
run_rnacocktail.py long_correct \
    --kmer 23 \
    --solid 3 \
    --short seq.fq.gz \
    --long seq_long.fa \
    --outdir out \
    --workdir work \
    --lordec /path/to/lordec-correct \
    --threads 4 \
    --sample A
```   

**Example 11 (long_align):**   
Run of RNACocktail for long-read alignment (STARlong).   
```
run_rnacocktail.py long_align \
    --long work/lordec/A/long_corrected.fa \
    --outdir out \
    --workdir work \
    --starlong /path/to/STARlong \
    --threads 4 \
    --sample A \
    --sam2psl /path/to/sam2psl.py \
    --samtools /path/to/samtools -\
    -genome_dir /path/to/STAR/genome_idx
```   
**Example 12 (long_reconstruct):**   

Run of RNACocktail for long-read transcriptome reconstruction (IDP).   
```
run_rnacocktail.py long_reconstruct \
    --alignment work/hisat2/A/alignments.sorted.bam \
    --short_junction work/hisat2/A/splicesites.bed \
    --long_alignment work/starlong/A/Aligned.out.psl \
    --outdir out \
    --workdir work \
    --idp /path/to/runIDP.py \
    --threads 4 \
    --sample A \
    --read_length 100 \
    --ref_genome genome.GRCh37.fa \
    --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd \
    --ref_gpd genes.GRCh37.refFlat.txt \
    --samtools /path/to/samtools \
    --idp_cfg idp.cfg
```    

**Example 13 (long_fusion):**   
Run of RNACocktail for long-read fusion detection (IDP-fusion).   
```
run_rnacocktail.py long_fusion \
    --alignment work/hisat2/A/alignments.sorted.bam \
    --short_junction work/hisat2/A/splicesites.bed \
    --short_fasta seq.fa \
    --long_fasta work/lordec/A/long_corrected.fa \
    --outdir out \
    --workdir work \
    --threads 4 \
    --sample A \
    --ref_genome genome.GRCh37.fa \
    --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd \
    --ref_gpd genes.GRCh37.refFlat.txt \
    --read_length 100 \
    --genome_bowtie2_idx genome.bt2_idx \
    --transcriptome_bowtie2_idx genes.bt2_idx \
    --uniqueness_bedgraph uniqueness.bedGraph \
    --gmap_idx gmap_idx \
    --idpfusion /path/to/runIDP.py \
    --samtools /path/to/samtools \
    --idpfusion_cfg idpfusion.cfg
```   

**Example 14 (variant):**   
Run of RNACocktail for RNA-Seq variant calling (GATK) with enabled IndelRealignment.
```
run_rnacocktail.py variant \
    --alignment work/hisat2/A/alignments.sorted.bam \
    --outdir out \
    --workdir work \
    --picard /path/to/picard.jar \
    --gatk /path/to/GenomeAnalysisTK.jar \
    --threads 10 \
    --sample A \
    --ref_genome genome.GRCh37.fa \
    --IndelRealignment \
    --knownsites dbsnp_138.b37.vcf
```    

**Example 15 (editing):**
Run of RNACocktail for RNA editing detection (GIREMI)
```
run_rnacocktail.py editing \
    --alignment work/gatk/A/bsqr.bam \
    --variant work/gatk/A/variants_filtered.vcf \
    --strand_pos test/GRCh37_strand_pos.bed \
    --genes_pos test/GRCh37_genes_pos.bed \
    --outdir out \
    --workdir work \
    --giremi_dir /path/to/giremi/directory/ \
    --gatk /path/to/GenomeAnalysisTK.jar \
    --samtools /path/to/samtools \
    --htslib_dir /path/to/htslib/directory/ \
    --threads 10 \
    --sample A \
    --ref_genome genome.GRCh37.fa \
    --knownsites dbsnp_138.b37.vcf
```   

**Example 16 (fusion):**
Run of RNACocktail for RNA fusion detection (FusionCatcher)
```
run_rnacocktail.py fusion \
    --data_dir /path/to/fusioncatcher/ensembl/data/directory/ \
    --input seq_1.fq.gz,seq_2.fq.gz \
    --outdir out \
    --workdir work \
    --fusioncatcher /path/to/fusioncatcher \
    --threads 4 \
    --sample A
```   

**Example 17 (all):**
Run all pipeline steps (Short-read example)
```
run_rnacocktail.py all \
    --outdir out \
    --workdir work \
    --threads 10 \
    --1 A1_1.fq.gz,A2_1.fq.gz B1_1.fq.gz,B2_1.fq.gz \
    --2 A1_2.fq.gz,A2_2.fq.gz B1_2.fq.gz,B2_2.fq.gz \
    --sample all_A1,all_A2 all_B1,all_B2 \
    --ref_gtf genes.GRCh37.gtf \
    --ref_genome genome.GRCh37.fa \
    --align_idx hisat2-idx \
    --quantifier_idx salmon_fmd_idx \
    --unzip \
    --file_format fastq.gz \
    --IndelRealignment \
    --CleanSam \
    --knownsites dbsnp_138.b37.vcf \
    --strand_pos test/GRCh37_strand_pos.bed \
    --genes_pos test/GRCh37_genes_pos.bed \
    --data_dir /path/to/fusioncatcher/ensembl/data/directory/ \
    --giremi_dir /path/to/giremi/directory/ \
    --gatk /path/to/GenomeAnalysisTK.jar \
    --htslib_dir /path/to/htslib/directory/ \
    --picard /path/to/picard.jar \
    --samtools /path/to/samtools \
    --hisat2 /path/to/hisat2 \
    --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
    --stringtie /path/to/stringtie \
    --salmon /path/to/salmon \
    --featureCounts /path/to/featureCounts \
    --oases /path/to/oases \
    --velveth /path/to/velveth \
    --velvetg /path/to/velvetg \
    --lordec /path/to/lordec-correct \
    --sam2psl /path/to/sam2psl.py \
    --fusioncatcher /path/to/fusioncatcher
```   


**Example 18 (all):**
Run all pipeline steps (long-read example)
```
run_rnacocktail.py all \
    --outdir out \
    --workdir work \
    --threads 10 \
    --U seq_short.fa \
    --long seq_long.fa \
    --sample all_C \
    --ref_gtf genes.GRCh37.gtf \
    --ref_genome genome.GRCh37.fa \
    --align_idx hisat2-idx \
    --quantifier_idx salmon_fmd_idx \
    --unzip \
    --file_format fasta \
    --IndelRealignment \
    --CleanSam \
    --knownsites dbsnp_138.b37.vcf \
    --strand_pos test/GRCh37_strand_pos.bed \
    --genes_pos test/GRCh37_genes_pos.bed \
    --data_dir /path/to/fusioncatcher/ensembl/data/directory/ \
    --giremi_dir /path/to/giremi/directory/ \
    --gatk /path/to/GenomeAnalysisTK.jar \
    --htslib_dir /path/to/htslib/directory/ \
    --star_genome_dir /path/to/STAR/genome_idx/ \
    --genome_bowtie2_idx genome.bt2_idx \
    --transcriptome_bowtie2_idx genes.bt2_idx \
    --uniqueness_bedgraph uniqueness.bedGraph \
    --gmap_idx gmap_idx \
    --ref_all_gpd hg19.all.refSeq_gencode_ensemble_EST_known.gpd \
    --ref_gpd genes.GRCh37.refFlat.txt \
    --read_length 100 \
    --picard /path/to/picard.jar \
    --hisat2_opts \"-f\" \
    --idp /path/to/idp/runIDP.py \
    --idpfusion /path/to/idpfusion/runIDP.py \
    --samtools /path/to/samtools \
    --hisat2 /path/to/hisat2 \
    --hisat2_sps /path/to/hisat2_extract_splice_sites.py \
    --stringtie /path/to/stringtie \
    --salmon /path/to/salmon \
    --featureCounts /path/to/featureCounts \
    --oases /path/to/oases \
    --velveth /path/to/velveth \
    --velvetg /path/to/velvetg \
    --lordec /path/to/lordec-correct \
    --sam2psl /path/to/sam2psl.py \
    --fusioncatcher /path/to/fusioncatcher
```  




**准备基因组的index文件**   
**HISAT2**   
`hisat2-build [options] reference.fa hisat2_index_basename`   
[more](http://ccb.jhu.edu/software/hisat2/manual.shtml)   

**Salmon-SMEM**   
`salmon index -t reference.fa -i salmon_index_basename --type fmd`   
[more](http://salmon.readthedocs.io/en/latest/salmon.html#using-salmon)   

**STAR**   
`STAR --runMode genomeGenerate --genomeDir STAR_index_basename --genomeFastaFiles reference.fa --runThreadN 4`      
[more](http://chagall.med.cornell.edu/RNASEQcourse/STARmanual.pdf)

**GMAP**   
`gmap_build -d gmap_index_basename reference.fa`   
[more](http://research-pub.gene.com/gmap/)

**bowtie2**   
`bowtie2-build reference.fa bowtie2_index_basename`   
[more](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)





**测试运行(简单版)**   

```
mkdir example_small
cd example_small
```

**Download reference genome (chromosome 21) FASTA file**   
```
wget ftp://ftp.ensembl.org/pub/release-75//fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.75.dna.chromosome.21.fa.gz
```

**Unzip reference genome (chromosome 21) FASTA file**   
```
gunzip Homo_sapiens.GRCh37.75.dna.chromosome.21.fa.gz
```   

**Download reference annotation GTF file**   
```
wget ftp://ftp.ensembl.org/pub/release-75//gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz
```   

**Unzip reference annotation GTF file**   
```
gunzip Homo_sapiens.GRCh37.75.gtf.gz
```

**Restrict GTF to chromosome 21**   
```
less Homo_sapiens.GRCh37.75.gtf |awk '{if ($1==21) print}' > Homo_sapiens.GRCh37.75.chromosome.21.gtf
```   


**Index genome with HISAT2**   
```
# conda  install hisat2
hisat2-build Homo_sapiens.GRCh37.75.dna.chromosome.21.fa Homo_sapiens.GRCh37.75.dna.chromosome.21.HISAT2
```   

**Test alignment step using HISAT2**   
```
run_rnacocktail.py align \
    --align_idx Homo_sapiens.GRCh37.75.dna.chromosome.21.HISAT2 \
    --outdir out \
    --workdir work \
    --ref_gtf Homo_sapiens.GRCh37.75.chromosome.21.gtf \
    --1 ../A1_1.fq.gz  \
    --2 ../A1_2.fq.gz \
    --hisat2 hisat2 \
    --hisat2_sps hisat2_extract_splice_sites.py  \
    --samtools samtools 
    --sample A
```   



**Test reconstruction step using StringTie**   
```
run_rnacocktail.py reconstruct \
    --alignment_bam work/hisat2/A/alignments.sorted.bam \
    --outdir out \
    --workdir work \
    --ref_gtf Homo_sapiens.GRCh37.75.chromosome.21.gtf \
    --stringtie stringtie \
    --sample A
```


**测试运行(docker版)**   

```
mkdir example
chmod -R 777 example/
cd example

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Initialization"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Download reference genome (chromosome 21) FASTA file"
wget ftp://ftp.ensembl.org/pub/release-90//fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz

echo "Unzip reference genome (chromosome 21) FASTA file"
gunzip Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz

echo "Download reference annotation GTF file"
wget ftp://ftp.ensembl.org/pub/release-90//gtf/homo_sapiens/Homo_sapiens.GRCh38.90.gtf.gz

echo "Unzip reference annotation GTF file"
gunzip Homo_sapiens.GRCh38.90.gtf.gz

echo "Restrict GTF to chromosome 21"
less Homo_sapiens.GRCh38.90.gtf |awk '{if ($1==21) print}' > Homo_sapiens.GRCh38.90.chromosome.21.gtf

echo "Prepare reference transcriptome FASTA file"
docker run -v=${PWD}/:/work_dir/ marghoob/rnacocktail gffread \
                                /work_dir/index/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                -g /work_dir/index/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                -w /work_dir/index/Homo_sapiens.GRCh38.cdna.21.fa

gunzip -c ../C_long.fa.gz > C_long.fa
gunzip -c ../C_short.fa.gz > C_short.fa
gunzip -c ../GRCh38_genes_pos.bed.gz > GRCh38_genes_pos.bed
gunzip -c ../GRCh38_strand_pos.bed.gz > GRCh38_strand_pos.bed
gunzip -c ../GRCh38.21.gpd.gz > GRCh38.21.gpd


echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test short-read alignment (HISAT2)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Index genome (chromosome 21) with HISAT2"
docker run -v=${PWD}/:/work_dir/ marghoob/rnacocktail hisat2-build \
                                /work_dir/index/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                /work_dir/index/Homo_sapiens.GRCh38.dna.chromosome.21.HISAT2
                                
                                
for sample in A1 A2 B1 B2
do
docker run -v=${PWD}/:/work_dir/ marghoob/rnacocktail run_rnacocktail.py align \
                                --align_idx /work_dir/index/Homo_sapiens.GRCh38.dna.chromosome.21.HISAT2 \
                                --outdir /work_dir/01_align \
                                --workdir /work_dir/work \
                                --ref_gtf /work_dir/Ref/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                --1 /work_dir/raw_data/${sample}_1.fq.gz  \
                                --2 /work_dir/raw_data/${sample}_2.fq.gz \
                                --sample ${sample}
done
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test short-read transcriptome reconstruction (StringTie)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
for sample in A1 A2 B1 B2
do
docker run -v=${PWD}/:/work_dir/ marghoob/rnacocktail run_rnacocktail.py reconstruct \
                                --alignment_bam /work_dir/01_align/hisat2/${sample}/alignments.sorted.bam \
                                --outdir /work_dir/02_reconstruct/  \
                                --workdir /work_dir/work \
                                --ref_gtf /work_dir/Ref/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                --sample ${sample}
done

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test quantification (Salmon-SMEM)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Index transcriptome with Salmon-SMEM"
docker run -v=${PWD}/:/work_dir/  marghoob/rnacocktail salmon index \
                                -t /work_dir/Ref/Homo_sapiens.GRCh38.cdna.21.fa  \
                                -i /work_dir/index/Homo_sapiens.GRCh38.cdna.21.Salmon.fmd \
                                --type fmd
for sample in A1 A2 B1 B2
do
docker run -v=${PWD}/:/work_dir/ marghoob/rnacocktail run_rnacocktail.py quantify \
                                --quantifier_idx /work_dir/index/Homo_sapiens.GRCh38.cdna.21.Salmon.fmd \
                                --1 /work_dir/raw_data/${sample}_1.fq.gz \
                                --2 /work_dir/raw_data/${sample}_2.fq.gz \
                                --libtype IU \
                                --salmon_k 19 \
                                --outdir /work_dir/03_quantify/ \
                                --workdir /work_dir/work \
                                --threads 4 \
                                --sample ${sample} \
                                --unzip
done

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test differential expression analysis based on alignment"
echo "results(DESeq2)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py diff \
                                --alignments /work_dir/example/work/hisat2/A1/alignments.sorted.bam,/work_dir/example/work/hisat2/A2/alignments.sorted.bam /work_dir/example/work/hisat2/B1/alignments.sorted.bam,/work_dir/example/work/hisat2/B2/alignments.sorted.bam \
                                --sample A1,A2 B1,B2 \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                --outdir /work_dir/example/out/diff-alignment/ \
                                --workdir /work_dir/example/work/diff-alignment/

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test differential expression analysis based on alignment-free"
echo "quantifications (DESeq2)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py diff \
                                --quant_files /work_dir/example/work/salmon_smem/A1/quant.sf,/work_dir/example/work/salmon_smem/A2/quant.sf /work_dir/example/work/salmon_smem/B1/quant.sf,/work_dir/example/work/salmon_smem/B2/quant.sf \
                                --sample A1,A2 B1,B2 \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                --outdir /work_dir/example/out/diff-quant \
                                --workdir /work_dir/example/work/diff-quant

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test de novo assembly (Oases)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py denovo \
                                --1 /work_dir/A1_1.fq.gz \
                                --2 /work_dir/A1_2.fq.gz \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 4 \
                                --sample A1 \
                                --file_format fastq.gz


echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test short-read fusion detection (FusionCatcher)"
echo "Note: downloading FusionCatcher necessary data may take 1-2 hours"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
mkdir fusioncatcher_data
cd fusioncatcher_data
wget https://sourceforge.net/projects/fusioncatcher/files/data/human_v90.tar.gz.aa &
wget https://sourceforge.net/projects/fusioncatcher/files/data/human_v90.tar.gz.ab &
wget https://sourceforge.net/projects/fusioncatcher/files/data/human_v90.tar.gz.ac
wget https://sourceforge.net/projects/fusioncatcher/files/data/human_v90.tar.gz.ad
cat human_v90.tar.gz.* | tar xz
rm -rf human_v90.tar.gz.a*

wget http://sourceforge.net/projects/fusioncatcher/files/test/reads_1.fq.gz &
wget http://sourceforge.net/projects/fusioncatcher/files/test/reads_2.fq.gz
cd ..

docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py fusion \
                                --data_dir /work_dir/example/fusioncatcher_data/human_v90/ \
                                --input /work_dir/example/fusioncatcher_data/reads_1.fq.gz,/work_dir/example/fusioncatcher_data/reads_2.fq.gz \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 4 \
                                --sample D

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test long-read error correction (LoRDEC)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py long_correct \
                                --kmer 23 \
                                --solid 3 \
                                --short /work_dir/example/C_short.fa \
                                --long /work_dir/example/C_long.fa \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --sample C

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test long-read alignment (STARlong)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Index genome with STAR"
mkdir STAR_genome_index_21/
chmod -R 777 STAR_genome_index_21/
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail STAR \
                                --runMode genomeGenerate \
                                --genomeDir /work_dir/example/STAR_genome_index_21/ \
                                --genomeFastaFiles /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                --runThreadN 4
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py long_align \
                                --long /work_dir/example/work/lordec/C/long_corrected.fa \
                                --threads 4 \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --sample C \
                                --genome_dir /work_dir/example/STAR_genome_index_21/

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test long-read transcriptome reconstruction (IDP)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Index genome with STAR"
echo "Download refFlat.txt annotation file"
wget http://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/refFlat.txt.gz
gunzip refFlat.txt.gz
echo "Restrict refFlat.txt annotation file to chromosome 21"
less refFlat.txt |grep chr21 |sed 's/chr21/21/g' > refFlat.21.txt
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py align \
                                --align_idx /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.HISAT2 \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh38.90.chromosome.21.gtf \
                                --1 /work_dir/C_short_1.fq.gz  \
                                --2 /work_dir/C_short_2.fq.gz \
                                --sample C
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py long_reconstruct \
                                --alignment /work_dir/example/work/hisat2/C/alignments.sorted.bam \
                                --short_junction /work_dir/example/work/hisat2/C/splicesites.bed \
                                --long_alignment /work_dir/example/work/starlong/C/Aligned.out.psl \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 4 \
                                --sample C \
                                --ref_genome /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                --ref_all_gpd /work_dir/example/GRCh38.21.gpd \
                                --ref_gpd /work_dir/example/refFlat.21.txt \
                                --read_length 101 \
                                --idp /opt/IDP/src/main/python/runIDP.py


echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test long-read fusion detection (IDP-fusion)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
wget https://www.healthcare.uiowa.edu/labs/au/IDP-fusion/files/MCF7_chr17q23-25chr20q13_Example.zip
unzip MCF7_chr17q23-25chr20q13_Example.zip
rm MCF7_chr17q23-25chr20q13_Example.zip
chmod 777 -R MCF7_chr17q23-25chr20q13_Example
echo "Download reference annotation GTF file for GRCh37"
wget ftp://ftp.ensembl.org/pub/release-75//gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz
echo "Unzip reference annotation GTF file"
gunzip Homo_sapiens.GRCh37.75.gtf.gz
echo "Restrict GTF to chromosome 17 and 20"
less Homo_sapiens.GRCh37.75.gtf |awk '{if ($1==20 || $1==17) print}'|sed 's/^/chr/' > Homo_sapiens.GRCh37.75.chromosome.17_20.gtf
echo "Index genome (chromosome 17 and 20) with HISAT2"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail hisat2-build \
                                /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/genome.chr17chr20.fasta \
                                /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/genome.chr17chr20.HISAT2
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py align \
                                --align_idx /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/genome.chr17chr20.HISAT2 \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh37.75.chromosome.17_20.gtf \
                                --U /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/short_reads.chr17q23-25chr20q13.tenth.fasta \
                                --sample F \
                                --hisat2_opts \"-f\"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py long_fusion \
                                --alignment /work_dir/example/work/hisat2/F/alignments.sorted.bam \
                                --short_junction /work_dir/example/work/hisat2/F/splicesites.bed \
                                --long_alignment /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/MCF7_ready_for_fusion_20140928.chr17q23-25chr20q13.sorted.psl \
                                --short_fasta /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/short_reads.chr17q23-25chr20q13.tenth.fasta \
                                --long_fasta /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/long_reads.chr17q23-25chr20q13.fasta \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 14 \
                                --sample F \
                                --ref_genome /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/genome.chr17chr20.fasta \
                                --ref_all_gpd /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/hg19.chr17chr20.gene_est.refFlat.txt \
                                --ref_gpd /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/refFlat_20140611.chr17chr20.gpd.txt \
                                --read_length 89 \
                                --genome_bowtie2_idx /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/bowtie2/bowtie2_index.chr17chr20 \
                                --transcriptome_bowtie2_idx /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/bowtie2/txn.chr17chr20 \
                                --uniqueness_bedgraph /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/wgEncodeDukeMapabilityUniqueness35bp.chr17chr20.bedGraph \
                                --gmap_idx /work_dir/example/MCF7_chr17q23-25chr20q13_Example/Data/gmap_index.chr17chr20/ \
                                --idpfusion /opt/IDP-fusion_1.1.1/bin/runIDP.py


echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test variant calling (GATK)"
echo "Note: the test on variant calling and RNA editing detection may take several hours"
echo "Note: GATK .jar file (GenomeAnalysisTK.jar) should be available in the test folder."
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Download reference genome FASTA file"
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/GRCh38_full_analysis_set_plus_decoy_hla.fa
echo "Download dbsnp files"
wget ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606/VCF/All_20170710.vcf.gz
gunzip All_20170710.vcf.gz
 awk '{if($0 !~ /^#/) print "chr"$0; else print $0}' All_20170710.vcf |sed  "s/chrMT/chrM/g" > All_20170710_chr.vcf
 mv All_20170710_chr.vcf All_20170710.vcf
echo "Index reference genome FASTA file"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail samtools faidx /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.fa
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail java -jar /usr/local/bin/picard.jar CreateSequenceDictionary \
                                R= /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.fa \
                                O= /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.dict
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail hisat2-build \
                                /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.fa \
                                /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.HISAT2
echo "Download NA12878 FASTQ files"

wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR896/SRR896663/SRR896663_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR896/SRR896663/SRR896663_2.fastq.gz

docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py align \
                                --align_idx /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.HISAT2 \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh38.90.gtf \
                                --1 /work_dir/example/SRR896663_1.fastq.gz  \
                                --2 /work_dir/example/SRR896663_2.fastq.gz \
                                --sample E \
                                --threads 10
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py variant \
                                --alignment /work_dir/example/work/hisat2/E/alignments.sorted.bam \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --picard /usr/local/bin/picard.jar \
                                --gatk /work_dir/GenomeAnalysisTK.jar \
                                --threads 10 \
                                --sample E \
                                --ref_genome /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.fa \
                                --IndelRealignment \
                                --CleanSam \
                                --knownsites /work_dir/example/All_20170710.vcf \
                                --picard /usr/local/bin/picard.jar

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test RNA editing detection (GIREMI)"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py editing \
                                --alignment /work_dir/example/work/gatk/E/bsqr.bam \
                                --variant /work_dir/example/work/gatk/E/variants_filtered.vcf \
                                --strand_pos /work_dir/example/GRCh38_strand_pos.bed \
                                --genes_pos /work_dir/example/GRCh38_genes_pos.bed \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --giremi_dir /usr/local/bin/ \
                                --gatk /work_dir/GenomeAnalysisTK.jar \
                                --htslib_dir /opt/htslib-1.3/ \
                                --threads 1 \
                                --sample E \
                                --ref_genome /work_dir/example/GRCh38_full_analysis_set_plus_decoy_hla.fa \
                                --knownsites /work_dir/example/All_20170710.vcf

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test run all pipeline (Short-read example)"
echo "This example should successfully run for short-read alignment,"
echo "reconstruction, quantification, differential expression,"
echo "denovo assembly, variant calling, and fusion detection."
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
cat <(less All_20170710.vcf |head -10000|grep "#") <(less All_20170710.vcf |awk '{if ($1=="chr21") print}') |sed "s/chr//g" > variants_21.vcf
cat GRCh38_genes_pos.bed |sed "s/chrM/MT/g"|sed "s/chr//g" > GRCh38_genes_pos_.bed
cat GRCh38_strand_pos.bed |sed "s/chrM/MT/g"|sed "s/chr//g" > GRCh38_strand_pos_.bed
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail samtools faidx /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail java -jar /usr/local/bin/picard.jar CreateSequenceDictionary \
                                R= /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                O= /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.dict
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py all \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 10 \
                                --1 /work_dir/A1_1.fq.gz,/work_dir/A2_1.fq.gz /work_dir/B1_1.fq.gz,/work_dir/B2_1.fq.gz \
                                --2 /work_dir/A1_2.fq.gz,/work_dir/A2_2.fq.gz /work_dir/B1_2.fq.gz,/work_dir/B2_2.fq.gz \
                                --sample all_A1,all_A2 all_B1,all_B2  \
                                --ref_gtf /work_dir/example/Homo_sapiens.GRCh38.90.chromosome.21.gtf  \
                                --ref_genome /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                --align_idx /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.HISAT2  \
                                --quantifier_idx /work_dir/example/Homo_sapiens.GRCh38.cdna.21.Salmon.fmd \
                                --unzip \
                                --file_format fastq.gz \
                                --IndelRealignment \
                                --CleanSam \
                                --knownsites /work_dir/example/variants_21.vcf \
                                --strand_pos /work_dir/example/GRCh38_strand_pos_.bed \
                                --genes_pos /work_dir/example/GRCh38_genes_pos_.bed \
                                --data_dir /work_dir/example/fusioncatcher_data/human_v90/ \
                                --giremi_dir /usr/local/bin/ \
                                --gatk /work_dir/GenomeAnalysisTK.jar \
                                --htslib_dir /opt/htslib-1.3/ \
                                --picard /usr/local/bin/picard.jar

echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
echo "Test run all pipeline (Long-read example)"
echo "Thid example should successfully run for short-read alignment,"
echo "reconstruction, denovo assembly, quantification, "
echo "variant calling and long-read error correction,"
echo "alignment, and reconstruction"
echo "--------------------------------------------------------"
echo "--------------------------------------------------------"
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail gmap_build \
                                -d /work_dir/example/gmap_chromosome.21 \
                                /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail bowtie2-build \
                                /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail bowtie2-build \
                                /work_dir/example/Homo_sapiens.GRCh38.cdna.21.fa \
                                /work_dir/example/Homo_sapiens.GRCh38.cdna.21
docker run -v=${PWD}/../:/work_dir/ marghoob/rnacocktail run_rnacocktail.py all \
                                --outdir /work_dir/example/out \
                                --workdir /work_dir/example/work \
                                --threads 10 \
                                --data_dir /work_dir/example/fusioncatcher_data/human_v90/ \
                                --U /work_dir/example/C_short.fa \
                                --long /work_dir/example/C_long.fa \
                                --sample all_C \
                                --ref_genome /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.fa \
                                --star_genome_dir /work_dir/example/STAR_genome_index_21/ \
                                --align_idx /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21.HISAT2 \
                                --quantifier_idx /work_dir/example/Homo_sapiens.GRCh38.cdna.21.Salmon.fmd \
                                --gmap_idx /work_dir/example/gmap_chromosome.21 \
                                --genome_bowtie2_idx /work_dir/example/Homo_sapiens.GRCh38.dna.chromosome.21 \
                                --transcriptome_bowtie2_idx /work_dir/example/Homo_sapiens.GRCh38.cdna.21 \
                                --ref_all_gpd /work_dir/example/GRCh38.21.gpd \
                                --ref_gpd /work_dir/example/refFlat.21.txt \
                                --file_format fasta \
                                --IndelRealignment \
                                --CleanSam \
                                --knownsites /work_dir/example/variants_21.vcf \
                                --strand_pos /work_dir/example/GRCh38_strand_pos_.bed \
                                --genes_pos /work_dir/example/GRCh38_genes_pos_.bed \
                                --read_length 101 \
                                --hisat2_opts \"-f\" \
                                --giremi_dir /usr/local/bin/ \
                                --gatk /work_dir/GenomeAnalysisTK.jar \
                                --htslib_dir /opt/htslib-1.3/ \
                                --picard /usr/local/bin/picard.jar \
                                --idp /opt/IDP/src/main/python/runIDP.py \
                                --idpfusion /opt/IDP-fusion_1.1.1/bin/runIDP.py

```




