<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="Hugo 0.60.1" />

  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="author" content="Qihao Wei" />
  <meta property="og:url" content="/cn/posts/alignment/" />
  <link rel="canonical" href="../../../cn/posts/alignment/" /><script type="application/ld+json">
  {
      "@context" : "http://schema.org",
      "@type" : "BlogPosting",
      "mainEntityOfPage": {
           "@type": "WebPage",
           "@id": "\/"
      },
      "articleSection" : "posts",
      "name" : "alignment",
      "headline" : "alignment",
      "description" : "Hisat2 HISAT2是一个快速和敏感的比对程序，用于将下一代测序读数（全基因组、转录组和外显组测序数据）映射到人类基因组群体（以及单个参考基因组）。基于BWT for graphs [Sirén et al. 2014]的方法，设计和实现的一种独创的方法graph FM index (GFM)。除了使用代表群体的的全局GFM的index外，HISAT2还使用局部GFM的index，他们共同覆盖整个基因组，每个index代表56Kbp的区域，一共有55000个index。这种新的索引策略称为层次graph FM index（HGFM）\nHISAT2中包含人类参考基因组和1230万个常见SNPs位点的index大小为6.2G。SNPs包括1100万个单核苷酸多态性、72.8万缺失和55.5万个插入。这个索引中的插入和删除片段都很小(通常小于20bp)。计划将结构变化(SV)合并到这个索引中。\nAlignment summary\nsummary for unpaired reads\n20000 reads; of these: 20000 (100.00%) were unpaired; of these: 1247 (6.24%) aligned 0 times 18739 (93.69%) aligned exactly 1 time 14 (0.07%) aligned \x26gt;1 times 93.77% overall alignment rate summary for paired reads\n10000 reads; of these: 10000 (100.00%) were paired; of these: 650 (6.50%) aligned concordantly 0 times 8823 (88.",
      "inLanguage" : "en-US",
      "author" : "Qihao Wei",
      "creator" : "Qihao Wei",
      "publisher": "Qihao Wei",
      "accountablePerson" : "Qihao Wei",
      "copyrightHolder" : "Qihao Wei",
      "copyrightYear" : "2018",
      "datePublished": "2018-12-12 00:00:00 \x2b0000 UTC",
      "dateModified" : "2018-12-12 00:00:00 \x2b0000 UTC",
      "url" : "\/cn\/posts\/alignment\/",
      "keywords" : [ "align","DNA","RNA-seq","HISAT2", ]
  }
</script>
<title>alignment - Example YinYong Blog</title>
  <meta property="og:title" content="alignment - Example YinYong Blog" />
  <meta property="og:type" content="article" />
  <meta name="description" content="Hisat2 HISAT2是一个快速和敏感的比对程序，用于将下一代测序读数（全基因组、转录组和外显组测序数据）映射到人类基因组群体（以及单个参考基因组）。基于BWT for graphs [Sirén et al. 2014]的方法，设计和实现的一种独创的方法graph FM index (GFM)。除了使用代表群体的的全局GFM的index外，HISAT2还使用局部GFM的index，他们共同覆盖整个基因组，每个index代表56Kbp的区域，一共有55000个index。这种新的索引策略称为层次graph FM index（HGFM）
HISAT2中包含人类参考基因组和1230万个常见SNPs位点的index大小为6.2G。SNPs包括1100万个单核苷酸多态性、72.8万缺失和55.5万个插入。这个索引中的插入和删除片段都很小(通常小于20bp)。计划将结构变化(SV)合并到这个索引中。
Alignment summary
summary for unpaired reads
20000 reads; of these: 20000 (100.00%) were unpaired; of these: 1247 (6.24%) aligned 0 times 18739 (93.69%) aligned exactly 1 time 14 (0.07%) aligned &gt;1 times 93.77% overall alignment rate summary for paired reads
10000 reads; of these: 10000 (100.00%) were paired; of these: 650 (6.50%) aligned concordantly 0 times 8823 (88." />

  <link rel="stylesheet" href="../../../css/flexboxgrid-6.3.1.min.css" />
  <link rel="stylesheet"
    href="../../../css/github-markdown.min.css" />
  <link rel="stylesheet" href="../../../css/highlight/tomorrow.min.css" />
  <link rel="stylesheet" href="../../../css/index.css">
  <link href="../../../index.xml" rel="alternate" type="application/rss+xml" title="Example YinYong Blog">
  
  <link href="https://fonts.googleapis.com/css?family=Arvo|Permanent+Marker" rel="stylesheet">
  
  

  
</head>


<body>
  <article class="post Chinese" id="article">
    <div class="row">
      <div class="col-xs-12 col-sm-10 col-md-8 col-sm-offset-1 col-md-offset-2 col-lg-6 col-lg-offset-3">
        <div class="site-header">
          
<header>
  <div class="signatures site-title">
    <a href="../../../cn/">Qihao Wei</a>
  </div>
</header>
<div class="row end-xs">
  
  
  <div class="lang-switch col-xs-3 col-xs-offset-9">
    <a href="../../../en/">English</a>
  </div>
  
  
</div>
<div class="header-line"></div>

        </div>
        <header class="post-header">
          <h1 class="post-title">alignment</h1>
          
          <div class="row post-desc">
            <div class="col-xs-6">
              
              <time class="post-date" datetime="2018-12-12 00:00:00 UTC">
                12 Dec 2018
              </time>
              
            </div>
            <div class="col-xs-6">
              
              <div class="post-author">
                <a target="_blank" href="../../../">@Qihao Wei</a>
              </div>
              
            </div>
          </div>
          
        </header>

        <div class="post-content markdown-body">
          


<div id="hisat2" class="section level2">
<h2>Hisat2</h2>
<p>HISAT2是一个快速和敏感的比对程序，用于将下一代测序读数（全基因组、转录组和外显组测序数据）映射到人类基因组群体（以及单个参考基因组）。基于<code>BWT for graphs [Sirén et al. 2014]</code>的方法，设计和实现的一种独创的方法<code>graph FM index (GFM)</code>。除了使用代表群体的的全局GFM的index外，HISAT2还使用局部GFM的index，他们共同覆盖整个基因组，每个index代表56Kbp的区域，一共有55000个index。这种新的索引策略称为<code>层次graph FM index（HGFM）</code></p>
<p>HISAT2中包含人类参考基因组和1230万个常见SNPs位点的index大小为6.2G。SNPs包括1100万个单核苷酸多态性、72.8万缺失和55.5万个插入。这个索引中的插入和删除片段都很小(通常小于20bp)。计划将结构变化(SV)合并到这个索引中。</p>
<p><strong>Alignment summary</strong></p>
<p>summary for unpaired reads</p>
<pre><code>20000 reads; of these:
  20000 (100.00%) were unpaired; of these:
    1247 (6.24%) aligned 0 times
    18739 (93.69%) aligned exactly 1 time
    14 (0.07%) aligned &gt;1 times
93.77% overall alignment rate</code></pre>
<p>summary for paired reads</p>
<pre><code>10000 reads; of these:
  10000 (100.00%) were paired; of these:
    650 (6.50%) aligned concordantly 0 times
    8823 (88.23%) aligned concordantly exactly 1 time
    527 (5.27%) aligned concordantly &gt;1 times
    ----
    650 pairs aligned concordantly 0 times; of these:
      34 (5.23%) aligned discordantly 1 time
    ----
    616 pairs aligned 0 times concordantly or discordantly; of these:
      1232 mates make up the pairs; of these:
        660 (53.57%) aligned 0 times
        571 (46.35%) aligned exactly 1 time
        1 (0.08%) aligned &gt;1 times
96.70% overall alignment rate</code></pre>
<p><strong>性能优化</strong></p>
<p><code>-p</code>选项使HISAT2启动指定数量的并行线程,每个线程在不同的处理器/核心上运行,使比对吞吐量大约增加为线程数量倍数。</p>
<p><strong>命令行</strong></p>
<p><strong>设置函数选项</strong></p>
<p>一些HISAT2选项可以指定函数，而不是单个数字或设置。在这些情况下，用户须指定三个参数：(a)函数类型<strong>F</strong>，(b)常数项<strong>B</strong>和(c)系数**A 。可用的函数类型：常数(C)、线性(L)、平方根(S)和自然对数(G)。参数指定为<code>F,B,A</code>-即，函数类型、常数项，系数用逗号分隔，没有空格。常数项和系数可以是负数和/或浮点数。</p>
<p>例如：<br />
函数选项指定为L，-0.4，-0.6，表示函数：<code>f(x) = -0.4 + -0.6 * x</code><br />
函数选项指定为G, 1, 5.4, 表示函数：<code>f(x) = 1.0 + 5.4 * ln(x)</code></p>
<p><strong>用法</strong><br />
<code>hisat2 [options]* -x &lt;hisat2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --sra-acc &lt;SRA accession number&gt;} [-S &lt;sam&gt;]</code></p>
</div>

        </div>
        

        

<div class="releated-content">
  <h3>Related Posts</h3>
  <ul>
    
    <li><a href="../../../cn/posts/quantification-transcript-level-salmon/">Quantification(transcript-level)--Salmon</a></li>
    
    <li><a href="../../../cn/posts/%E5%AD%A6%E4%B9%A0%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%B8%B8%E7%94%A8%E6%B5%81%E7%A8%8B-1/">学习生物信息常用流程-1</a></li>
    
  </ul>
</div>


        
        
        <div style="height: 50px;"></div>
        
        <div class="post-comments">
          <div id="disqus_thread"></div>
<script>
  window.addEventListener("load", () => {
    (function() {
      
      var d = document,
        s = d.createElement("script");
      s.src = "https://joway.disqus.com/embed.js";
      s.setAttribute("data-timestamp", +new Date());
      (d.head || d.body).appendChild(s);
    })();
  });
</script>
<noscript
  >Please enable JavaScript to view the
  <a href="https://disqus.com/?ref_noscript"
    >comments powered by Disqus.</a
  ></noscript
>

        </div>
        
        

        <div class="site-footer">
  
  <div class="site-footer-item">
    <a href="../../../index.xml" target="_blank">RSS</a>
  </div>
  
  <div class="site-footer-item">
    <a href="https://joway.io" target="_blank">About</a>
  </div>
  
  
</div>

      </div>
    </div>
  </article>

  <script src="../../../js/highlight.pack.js"></script>


<script>
  hljs.initHighlightingOnLoad();
  
  
  
    
    
  
</script>

  

</body>

</html>