---
title: 使用R下载SRA数据
author: ''
date: '2018-12-26'
slug: 使用r下载sra数据
categories:
  - R
  - Bioinfomatics
tags:
  - data download
css: []
description: ''
highlight: yes
index: no
scripts: []
---

```{r echo=TRUE, eval=FALSE, include=TRUE}
library(SRAdb)
setwd("/home/wqh/sra_download/SRR6660814")
sra_dbname <- "../SRAmetadb.sqlite"
sra_con <- dbConnect(dbDriver("SQLite"), sra_dbname)

# For Failed to open TCP Connection
# iptables -I INPUT -p tcp --dport 33001 -j ACCEPT
# iptables -I OUTPUT -p tcp --dport 33001 -j ACCEPT


## Download sra files from NCBI SRA using ftp protocol:
getSRAfile( c("SRR6660815"), sra_con, fileType='fastq')


## Download fastq files from EBI  ftp siteusing fasp protocol:
ascpCMD <-  '/home/wqh/.aspera/connect/bin/ascp -QT -l 300m  -P 33001 -i /home/wqh/.aspera/connect/etc/asperaweb_id_dsa.openssh'
s1 <- Sys.time()
getSRAfile(c("SRR6660814"), sra_con = sra_con, fileType = "fastq", srcType = "fasp",ascpCMD = ascpCMD )
s2 <- Sys.time()



## Start IGV from R if no IGV running
## Not run: startIGV(memory='mm')

## load BAM files to IGV
## Not run: 
exampleBams = file.path(system.file('extdata',package='SRAdb'), dir(system.file('extdata',package='SRAdb'),pattern='bam$'))
sock <- IGVsocket()
IGVload(sock,exampleBams)


## End(Not run)
## Change the IGV genome
## Not run: 
IGVgenome(sock,genome='hg18')

## End(Not run)
## Go to a specified region in IGV
## Not run: 
IGVgoto(sock,'chr1:1-10000')
IGVgoto(sock,'TP53')

## End(Not run)

## Make a snapshot of the current IGV window
## Not run:   
IGVsnapshot(sock)
dir()

## End(Not run)


## create a graphNEL object from SRA accessions, which are full text search results of terms 'primary thyroid cell line'
g <- sraGraph('primary thyroid cell line', sra_con)

## Not run: 
library(Rgraphviz)
attrs <- getDefaultAttrs(list(node=list(fillcolor='lightblue', shape='ellipse')))
plot(g, attrs=attrs)

## End(Not run)

dbDisconnect(sra_con) 

} else {
  print("use getSRAdbFile() to get a copy of the SRAmetadb.sqlite file
and then rerun the example")
}

```



