---
title: 使用R下载SRA数据
author: ''
date: '2018-12-26'
slug: 使用r下载sra数据
categories:
  - R
  - Bioinfomatics
tags:
  - data download
css: []
description: ''
highlight: yes
index: no
scripts: []
---



<pre class="r"><code>library(SRAdb)
setwd(&quot;/home/wqh/sra_download/SRR6660814&quot;)
sra_dbname &lt;- &quot;../SRAmetadb.sqlite&quot;
sra_con &lt;- dbConnect(dbDriver(&quot;SQLite&quot;), sra_dbname)

# For Failed to open TCP Connection
# iptables -I INPUT -p tcp --dport 33001 -j ACCEPT
# iptables -I OUTPUT -p tcp --dport 33001 -j ACCEPT


## Download sra files from NCBI SRA using ftp protocol:
getSRAfile( c(&quot;SRR6660815&quot;), sra_con, fileType=&#39;fastq&#39;)


## Download fastq files from EBI  ftp siteusing fasp protocol:
ascpCMD &lt;-  &#39;/home/wqh/.aspera/connect/bin/ascp -QT -l 300m  -P 33001 -i /home/wqh/.aspera/connect/etc/asperaweb_id_dsa.openssh&#39;
s1 &lt;- Sys.time()
getSRAfile(c(&quot;SRR6660814&quot;), sra_con = sra_con, fileType = &quot;fastq&quot;, srcType = &quot;fasp&quot;,ascpCMD = ascpCMD )
s2 &lt;- Sys.time()



## Start IGV from R if no IGV running
## Not run: startIGV(memory=&#39;mm&#39;)

## load BAM files to IGV
## Not run: 
exampleBams = file.path(system.file(&#39;extdata&#39;,package=&#39;SRAdb&#39;), dir(system.file(&#39;extdata&#39;,package=&#39;SRAdb&#39;),pattern=&#39;bam$&#39;))
sock &lt;- IGVsocket()
IGVload(sock,exampleBams)


## End(Not run)
## Change the IGV genome
## Not run: 
IGVgenome(sock,genome=&#39;hg18&#39;)

## End(Not run)
## Go to a specified region in IGV
## Not run: 
IGVgoto(sock,&#39;chr1:1-10000&#39;)
IGVgoto(sock,&#39;TP53&#39;)

## End(Not run)

## Make a snapshot of the current IGV window
## Not run:   
IGVsnapshot(sock)
dir()

## End(Not run)


## create a graphNEL object from SRA accessions, which are full text search results of terms &#39;primary thyroid cell line&#39;
g &lt;- sraGraph(&#39;primary thyroid cell line&#39;, sra_con)

## Not run: 
library(Rgraphviz)
attrs &lt;- getDefaultAttrs(list(node=list(fillcolor=&#39;lightblue&#39;, shape=&#39;ellipse&#39;)))
plot(g, attrs=attrs)

## End(Not run)

dbDisconnect(sra_con) 

} else {
  print(&quot;use getSRAdbFile() to get a copy of the SRAmetadb.sqlite file
and then rerun the example&quot;)
}</code></pre>
